public/app/features/gops/configuration-tracker/hooks/irmHooks.ts

```diff
@@ -0,0 +1,380 @@
+import { useEffect, useState } from 'react';
+
+import { getBackendSrv, locationService } from '@grafana/runtime';
+import { alertRuleApi } from 'app/features/alerting/unified/api/alertRuleApi';
+import { alertmanagerApi } from 'app/features/alerting/unified/api/alertmanagerApi';
+import { onCallApi } from 'app/features/alerting/unified/api/onCallApi';
+import { usePluginBridge } from 'app/features/alerting/unified/hooks/usePluginBridge';
+import { SupportedPlugin } from 'app/features/alerting/unified/types/pluginBridges';
+import { GRAFANA_RULES_SOURCE_NAME } from 'app/features/alerting/unified/utils/datasource';
+import { createUrl } from 'app/features/alerting/unified/utils/url';
+import { Receiver } from 'app/plugins/datasource/alertmanager/types';
+
+interface UrlLink {
+  url: string;
+  queryParams?: Record<string, string>;
+}
+export interface StepButtonDto {
+  type: 'openLink' | 'dropDown';
+  label: string;
+  labelOnDone?: string;
+  done?: boolean;
+  urlLink?: UrlLink; // only for openLink
+  urlLinkOnDone?: UrlLink; // only for openLink
+  options?: Array<{ label: string; value: string }>; // only for dropDown
+  onClickOption?: (value: string) => void; // only for dropDown
+}
+export interface SectionDtoStep {
+  title: string;
+  description: string;
+  button: StepButtonDto;
+}
+export interface SectionDto {
+  title: string;
+  description: string;
+  steps: SectionDtoStep[];
+}
+export interface SectionsDto {
+  sections: SectionDto[];
+}
+
+function isCreateAlertRuleDone() {
+  const { data: namespaces = [] } = alertRuleApi.endpoints.prometheusRuleNamespaces.useQuery(
```

Comment:
It might be better to use `rulerRules` instead of Prometheus
prometheus endpoint response is usually delayed (due to eventual consistency), so it might take even a few minutes until a new rule will be returned by that endpoint